src/main/java/org/jenkinsci/plugins/workflow/cps/DSL.java

```diff
@@ -217,23 +219,23 @@ protected Object invokeStep(StepDescriptor d, Object args) {
      * @param args The arguments passed to the step.
      */
     protected Object invokeStep(StepDescriptor d, String name, Object args) {
-        final NamedArgsAndClosure ps = parseArgs(args, d);
-
-        CpsThread thread = CpsThread.current();
-
-        FlowNode an;
 
         // TODO: generalize the notion of Step taking over the FlowNode creation.
         boolean hack = d instanceof ParallelStep.DescriptorImpl || d instanceof LoadStep.DescriptorImpl;
 
-        if (ps.body == null && !hack) {
+        FlowNode an = null;
+        CpsThread thread = CpsThread.current();
+        if (!d.takesImplicitBlockArgument() && !hack) {
```

Comment:
Yikes, good catch. Yeah, so in the case of a step that does not require a body, like `echo`, the `takesImplicitBlockArgument` will make it create  a `StepAtomNode`. However, for the `timeout` step, I had created it without a body and it actually threw an `NPE` instead of `java.lang.IllegalStateException: There is no body to invoke` in the original code. I suspect you meant a different test condition though?

Reply:
empty body check in 22c4ae3b978a397ec02c5c4131ac54bac95a07d8